tests/kernel/sched/schedule_api/src/test_sched_priority.c

```diff
@@ -95,6 +111,63 @@ void test_priority_preemptible(void)
 	k_thread_priority_set(k_current_get(), old_prio);
 }
 
+/**
+ * @brief Validate scheduling sequence of preemptive threads with start delay
+ *
+ * @details Create three preemptive threads which are of priority
+ * higher than current thread. Make sure that the highest priority
+ * and longest waiting thread is scheduled first.
+ *
+ * @ingroup kernel_sched_tests
+ */
+void test_priority_preemptible_wait_prio(void)
+{
+	int old_prio = k_thread_priority_get(k_current_get());
+	k_tid_t tid[THREAD_NUM];
+	uint8_t tid_chk[4] =  { 0, 1, 2, 3 };
+
+	k_sem_init(&sync_sema, 0, THREAD_NUM);
+
+	/* set current thread to a non-negative priority */
+	last_prio = 2;
+	k_thread_priority_set(k_current_get(), last_prio);
+
```

Comment:
This is a little racy.  You don't know that a millisecond boundary will not happen between these thread creation calls, throwing your expected math off.  Put a `k_usleep(1)` call here (or anything else that will block on a timeout), that will ensure that this code starts running at the start of a system tick and minimize the probability of a timing glitch.

Alternatively, there is now a facility for absolute/uptime timeouts (e.g `K_TIMEOUT_ABS_*()`) which you can use instead and which will guarantee to fire at the time you expect without the possibility of race conditions vs. real time.

Reply:
Thank you for your valuable advice.

Reply:
@yingmingx , have you addressed Andy's comment?

Reply:
@andyross . Hi, Andy, I have addressed this comment. Please take a review, thanks.